src/meson.build

```diff
@@ -0,0 +1,117 @@
+project('bpp', ['cpp','c'],
+        version: '4.4.1',
+        default_options : [
+          'optimization=3',     # -O3
+          'warning_level=1',    # -Wall
+          'buildtype=release',  # specify -Ddebug=true on commmand line for -g
+          'b_ndebug=if-release'
+        ],
+        meson_version: '>= 0.56',
+        license: 'AGPLv3')
+
+cc = meson.get_compiler('c')
+
+warn_flags = ['-Wsign-compare']
+add_project_arguments(['-D_GNU_SOURCE'] + cc.get_supported_arguments(warn_flags), language: 'c')
+
+avx_sources  = ['core_partials_avx.c' , 'core_likelihood_avx.c' ]
+avx2_sources = ['core_partials_avx2.c', 'core_likelihood_avx2.c']
+
+if host_machine.cpu_family().startswith('x86')
+  if get_option('have_sse3')
+    add_project_arguments(['-DHAVE_SSE3','-msse3'], language: 'c')
+  endif
+
+  if get_option('have_avx')
+    add_project_arguments(['-DHAVE_AVX'], language: 'c')
+  endif
+
+  if get_option('have_avx2')
+    add_project_arguments(['-DHAVE_AVX2'], language: 'c')
+  endif
+
+  if get_option('have_avx')
+    libavx  = static_library('avx' , avx_sources,  c_args: ['-mavx'])
+  else
+    libavx = dependency('', required: false)
+  endif
+
+  if get_option('have_avx2')
+    libavx2 = static_library('avx2', avx2_sources, c_args: ['-mavx2','-mfma'])
+  else
+    libavx2 = dependency('', required: false)
+  endif
+endif
```

Comment:
Would https://mesonbuild.com/Simd-module.html be useful here?

Reply:
That seems like a great way to choose which libraries to compile depending on what the compiler can handle.
Does this work for code that requires TWO options, such as _both_ `-mavx2` and `-mfma`?

Reply:
The initial key is for the option that will be checked. You can use a `simd_mod.check()` for a single compiler optimization, i.e. avx2, and pass the additional option along... `check()` will then promptly optimize out to empty if it isn't supported, which helps you avoid boilerplate defines/conditionals.

Reply:
Hmm, so something like this?

```
cdata = configuration_data()

simd = import('unstable-simd')
sse3_avx = simd.check('sse3_avx',
                      sse3: ['core_partials_sse.c','core_likelihood_sse.c'],
                      avx:  ['core_partials_avx.c','core_likelihood_avx.c'],
                      compiler: cc)

avx2 = simd.check('avx', avx2: ['core_partials_avx2.c','core_likelihood_avx2.c'], compiler: cc, c_args: ['-mfma'])
simdlibs = [sse3_avx[0], avx2[0]]
cdata.merge_from(avx2[1])
cdata.merge_from(sse3_avx[1])

configure_file(output : 'simdconfig.h', configuration : cdata)
```

I wonder if it would be possible to make this a bit less verbose.

Reply:
The simd module is currently unstable in part because we haven't got a lot of feedback about how people use it, so it's entirely possible it can be improved.

API design suggestions are welcome, as are suggestions for missing functionality.